securedrop/p2p.py

```diff
@@ -0,0 +1,140 @@
+import os
+import zlib
+from base64 import b64encode, b64decode
+from math import ceil
+from multiprocessing import shared_memory
+
+from securedrop import ClientBase, ServerBase
+from securedrop.file_transfer_packets import FILE_TRANSFER_P2P_CHUNK_SIZE, FileTransferP2PChunkPackets, \
+    FileTransferP2PFileInfoPackets, FILE_TRANSFER_P2P_FILEINFO_PACKETS_NAME, FILE_TRANSFER_P2P_CHUNK_PACKETS_NAME, \
+    FILE_TRANSFER_P2P_SENTINEL_PACKETS_NAME, FileTransferP2PSentinelPackets
+from securedrop.status_packets import StatusPackets
+from securedrop.utils import sha256_file
+
+
+class P2PClient(ClientBase):
+    def __init__(self, port, token, in_filename, in_file_size, in_file_sha256, progress_shm_name, progress_lock):
+        super().__init__("localhost", port)
+        self.token, self.in_filename, self.in_file_size, self.in_file_sha256, self.progress_shm_name, self.progress_lock = token, in_filename, in_file_size, in_file_sha256, progress_shm_name, progress_lock
```

Comment:
This line is super long. I am assuming the formatter will take care of it but thought I would point it out in case it doesn't.